### PR TITLE
Speedup the fanout unit tests

### PR DESCRIPTION
**Pull request by denis-tingajkin**
Signed-off-by: denis-tingajkin <denis.tingajkin@xored.com>

## Motivation and Context
Fanout unit tests runs 50-55s on sanity check stage. This PR reduce time to 15s.

## How Has This Been Tested?

- [ ] Covered by existing integration testing
- [ ] Added integration testing to cover
- [ ] Tested locally
- [ ] Have not tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: utils/helper/errtools/combine.go [hunk 1]
```diff
@@ -1,34 +1,48 @@
 package errtools
 
 import (
-	"errors"
 	"fmt"
 	"strings"
 )
 
 //Combine means collect a array of errors to the error
 func Combine(errs ...error) error {
```
Comment (lobkovilya) [inline utils/helper/errtools/combine.go]: Please consider using [pkg/errors](https://github.com/pkg/errors)

File: k8s/cmd/nsm-coredns/plugin/fanout/health.go [hunk 2]
```diff
@@ -4,53 +4,46 @@ import (
...
 
 func (h *dnsHealthClient) SetTLSConfig(cfg *tls.Config) {
 	h.c.Net = tcptlc
 	h.c.TLSConfig = cfg
 }
 
-func (h *dnsHealthClient) Check(p *dnsClient) error {
-	err := h.dnsPing(p.addr)
+func (h *dnsHealthClient) Check() error {
+	err := h.askAny(h.addr)
```
[the remark this answers is not in the document]
Comment (denis-tingajkin) [reply k8s/cmd/nsm-coredns/plugin/fanout/health.go]: thanks, fixed

File: k8s/cmd/nsm-coredns/plugin/fanout/fanout.go [hunk 3]
```diff
@@ -62,37 +62,31 @@ func (f *Fanout) Name() string {
...
 			}
-		}()
-	}
-	var first connectResult
-	select {
-	case <-time.After(defaultTimeout):
-		return dns.RcodeServerFailure, errNoHealthy
-	case first = <-result:
-		break
+			first = &conn
```
[the remark this answers is not in the document]
Comment (denis-tingajkin) [reply k8s/cmd/nsm-coredns/plugin/fanout/fanout.go]: thanks, fixed

File: k8s/cmd/nsm-coredns/plugin/fanout/fanout.go [hunk 4]
```diff
@@ -62,37 +62,31 @@ func (f *Fanout) Name() string {
...
-				if err != nil && attempt+1 < f.maxFailCount {
-					client.healthCheck()
+		go connect(client, req, result, f.maxFailCount)
+	}
+	var first *connectResult
+	for first == nil {
+		select {
+		case <-ctx.Done():
+			return dns.RcodeServerFailure, errContextDone
+		case <-time.After(defaultTimeout):
```
[the remark this answers is not in the document]
Comment (denis-tingajkin) [reply k8s/cmd/nsm-coredns/plugin/fanout/fanout.go]: thanks, fixed

File: utils/helper/errtools/combine.go (hunk 1 above)
Comment (denis-tingajkin) [reply utils/helper/errtools/combine.go]: thanks, fixed

**Pull request merged**